src/components/MainNav/MainNav.js

```diff
@@ -65,6 +66,8 @@ class MainNav extends Component {
 
   componentDidMount() {
     let curQuery = getLocationQuery(this.props.location);
+    const prevQueryState = {};
```

Comment:
Is there a risk of this value getting reinitialized out when you would want the values to persist? If so, considered using `this.prevQueryState` instead.

Reply:
The `componentDidMount` should only execute once when the `MainNav` is ready so we should be fine here.